src/clj/witan/send/goal_seek.clj

```diff
@@ -80,59 +94,133 @@
+                             (throw (ex-info "Baseline population is too low, can't modify"
+                                             {:population result}))
+                             (= result target-pop-median)
+                             (throw (ex-info "Baseline population already matches target population"
+                                             {:baseline-population result
+                                              :target-population target-pop-median}))
+                             :else
+                             result))]
+    (println (str "Baseline population for " target-year ": " baseline-pop))
+    (println "Modifying:" m)
+    (with-open [writer (io/writer (io/output-stream (str results-path (create-log-filename m))))]
+      (csv/write-csv writer (into [["modifier" "population"]] (map #(into [%1 %2]) [1] [baseline-pop])))
+      (loop [modifier (math/round (/ target-pop-median baseline-pop))
+             tested-modifiers [modifier]
+             population []]
+        (let [config (first (generate-configs base-config m modifier (+ modifier 1) step))
+              [projection pop-result] (target-result config
+                                                     target-year
+                                                     (first (get-in config [:transition-parameters :transitions-to-change])))]
+          (csv/write-csv writer [[pop-result modifier]])
+          (cond
+            (target-pop-to-high? pop-result target-pop-range)
+            (let [new-modifier (if (<= (apply min tested-modifiers)
+                                       (math/round (* modifier (- (/ pop-result target-pop-median) 1))))
+                                 (math/round (* modifier (- (/ pop-result target-pop-median) 1)))
+                                 (math/round (* modifier (- (/ pop-result target-pop-median) 0.5))))]
+              (recur new-modifier
+                     (into tested-modifiers [new-modifier])
+                     (into population [pop-result])))
+
+            (target-pop-to-low? pop-result target-pop-range)
+            (let [new-modifier (if (<= (apply max tested-modifiers)
+                                       (math/round (* modifier (+ (/ pop-result target-pop-median) 1))))
+                                 (math/round (* modifier (+ (/ pop-result target-pop-median) 1)))
+                                 (math/round (* modifier (+ (/ pop-result target-pop-median) 0.5))))]
+              (recur new-modifier
+                     (into tested-modifiers [new-modifier])
+                     (into population [pop-result])))
+
+            (within-pop-range? target-pop-range pop-result)
+            (let [config (update-results-path config results-path)]
+              (so/output-send-results projection (:output-parameters config))
+              (main/save-runtime-config config)
+              (main/save-runtime-metadata config (md/metadata config))
+              )))))))
```

Comment:
This will do for now, but feels a bit brittle. We should look into this in future versions.

Reply:
What are your issues? Too big? `cond` to complicated?

Reply:
streaming out the csv like this would mean we might lose it all if there was an exception or other error

Reply:
ah okay, so interestingly I wonder if this is actually causing an issue. My run last night found the best value and output the results but then the process never ended, I wonder if it was because the stream was still open. Do you have any immediate suggestions for how to improve this?